external-crates/move/crates/move-compiler/src/typing/translate.rs

```diff
@@ -816,22 +827,23 @@ fn variant_def(
     core::solve_constraints(context);
 
     for (_field_loc, _field_, idx_ty) in field_map.iter_mut() {
-        expand::type_(context, &mut idx_ty.1);
+        let (_idx, (_doc, ty)) = idx_ty;
+        expand::type_(context, ty);
     }
     field_map
         .into_iter()
-        .map(|(_, _, idx_ty)| idx_ty.clone())
+        .map(|(_, _, (idx, (_, ty)))| (*idx, ty.clone()))
         .collect::<Vec<_>>()
 }
 
 fn check_type_params_usage(
     context: &mut Context,
     type_parameters: &[N::DatatypeTypeParameter],
-    field_map: &Fields<Type>,
+    field_map: &Fields<(DocComment, Type)>,
 ) {
     let has_unresolved = field_map
         .iter()
-        .any(|(_, _, ty)| has_unresolved_error_type(&ty.1));
+        .any(|(_, _, (_, (_, ty)))| has_unresolved_error_type(ty));
```

Comment:
Would it make sense to define some `FieldType { ty: Type, comment: DocComment }` just to ease some of these pervasive pattern-matches? Then we could say:
```
    let has_unresolved = field_map
        .iter()
        .any(|(_, _, (_, fty))| has_unresolved_error_type(fty.ty));
``` 

I appreciate it may make other things more-cumbersome

Reply:
I tried but it felt strange for just the pair here. I wanted to have the index in there as well, but it became too strange with some of the other usages of `Fields`. 

I went through and at least tried to give names to the `_`s in these cases